Replicator/tests/ReplicatorCollectionTest.cc

```diff
@@ -178,87 +178,167 @@ TEST_CASE_METHOD(ReplicatorCollectionTest, "Use Zero Collections", "[Push][Pull]
     _expectedError = {LiteCoreDomain, kC4ErrorInvalidParameter};
     runPushPullReplication({}, {});
 }
+#endif
+
+#define DISABLE_PUSH_AND_PULL
+#define DISABLE_CONTINUOUS
+
+static std::set<string> getDocInfos(C4Database* db, C4CollectionSpec coll) {
+    std::set<string> ret;
+    C4Collection* collection = ReplicatorCollectionTest::getCollection(db, coll);
+    auto e = c4coll_enumerateAllDocs(collection, nullptr, ERROR_INFO());
+    {
+        TransactionHelper t(db);
+        while (c4enum_next(e, ERROR_INFO())) {
+            C4DocumentInfo info;
+            c4enum_getDocumentInfo(e, &info);
+            alloc_slice docID(info.docID);
+            alloc_slice revID(info.revID);
+            string entry = docID.asString()+"/"+revID.asString();
+            ret.insert(entry);
+        }
+    }
+    c4enum_free(e);
+    return ret;
+}
+
+
+struct CheckDBEntries {
+    CheckDBEntries(C4Database* db, C4Database* db2, vector<C4CollectionSpec> specs)
+    : _db(db)
+    , _db2(db2)
+    {
+        for (auto i = 0; i < specs.size(); ++i) {
+            _collSpecs.push_back(specs[i]);
+            _dbBefore.push_back(getDocInfos(_db, specs[i]));
+            _db2Before.push_back(getDocInfos(_db2, specs[i]));
+        }
+    }
+    
+    ~CheckDBEntries() {
+        vector<set<string>> dbAfter;
+        vector<set<string>> db2After;
+        for (auto i = 0; i < _collSpecs.size(); ++i) {
+            dbAfter.push_back(getDocInfos(_db, _collSpecs[i]));
+            db2After.push_back(getDocInfos(_db2, _collSpecs[i]));
+            CHECK(dbAfter[i].size() == _dbBefore[i].size());
+            for (auto& doc : _dbBefore[i]) {
+                CHECK(db2After[i].erase(doc) == 1);
+            }
+            for (auto& doc : _db2Before[i]) {
+                CHECK(db2After[i].erase(doc) == 1);
+            }
+            REQUIRE(db2After[i].size() == 0);
+        }
+    }
+    
+    C4Database* _db;
+    C4Database* _db2;
+    vector<C4CollectionSpec> _collSpecs;
+    vector<set<string>> _dbBefore;
+    vector<set<string>> _db2Before;
+};
+
 
 TEST_CASE_METHOD(ReplicatorCollectionTest, "Sync with Default Collection", "[Push][Pull]") {
     addCollDocs(db, Default, 10);
     addCollDocs(db2, Default, 10);
     
     SECTION("PUSH") {
+        CheckDBEntries check(db, db2, {Default});
```

Comment:
It seems unnecessary to check as each test starts with all empty collections. If adding docs fails, test will fail. 

If you really think the test should check, move it outside SECTION as they are all the same.

Reply:
I just realized your logic in CheckDBEntries.